src/routes/hydra.ts

```diff
@@ -1,162 +1,209 @@
-              .then((hydraResponse: any) => {
-              // All we need to do is to confirm that we indeed want to log in the user.
-              
-              // All we need to do now is to redirect the user back to hydra!
-              res.redirect(hydraResponse.redirect_to)
-            });
-          } else if (kratos_session){
-            // Figuring out the user
-            return kratosPublicEndpoint
-              // We need to know who the user is for hydra
-              .whoami(req as { headers: { [name: string]: string } })
-              .then( ({body, response} ) => {
-                // We need to get the email of the user. We don't want to do that via traits as
-                // they are dynamic. They would be part of the PublicAPI. That's not true
-                // for identity.addresses So let's get it via the AdmninAPI
-                const subject = body.identity.id
-                kratosAdminEndpoint
-                  .getIdentity(body.identity.id) 
-                  .then( ({response}) => {
-                    // User is authenticated, accept the LoginRequest and tell Hydra
-                    let acceptLoginRequest : AcceptLoginRequest = new AcceptLoginRequest()
-                    // We should probably use "id" here but the gitlab-scenario only works via "email"
-                    acceptLoginRequest.subject = subject
-                    acceptLoginRequest.context = (response as any).body
-   
-                    //acceptLoginRequest.subject = (body.identity.traits as { email: string }).email || "fallback@mail.com"
-                    
-                    return hydraAdminEndpoint.acceptLoginRequest(challenge, acceptLoginRequest)
-                      .then((hydraResponse: any) => {
-                        // All we need to do now is to redirect the user back to hydra!
-                        res.redirect(hydraResponse.body.redirectTo)
-                    })
-                  })
-
-              })
-          } else {
-            logger.error("Request and challenge are both set but user is not authenticated. Unknown state!")
-            res.status(400).send('Request and challenge are bot set but user is not authenticated. Please try again!');
-            next()
-          }
-        })
-        .catch((err) => {
-          // Something went wrong with validating hydra's challenge getting the LoginRequest
-          logger.error(err)
-          next()
-        });
-    }
+    logger.debug('hydraChallenge exists: ' + hydraChallenge)
+  }
+
+  if (config.securityMode !== SECURITY_MODE_STANDALONE) {
+    next(
+      new Error(
+        'Interaction with ORY Hydra only works in security standalone mode right now.'
+      )
+    )
+    return
+  }
+  const kratosSessionCookie = req.cookies.ory_kratos_session
+
+  if (kratosRequest) {
+    // FIXME @k9ert this state is undefined.
+    next(new Error('what should be done here?'))
+    return
+  }
+
+  if (!kratosSessionCookie) {
```

Comment:
Because we checked if `kratosRequest` is set, and returned, this can only be reached when `kratosRequest` is not set, which was the logic prior.

```
if (!request) {
  if (!kratos_session) {
```

is equal to (and easier to read as)

```
if (request) {
  return
}

if (!kratos_session) {
```

This pattern is called ["early return"](https://dev.to/yvonnickfrin/devtips-use-early-returns-to-avoid-nested-conditions-4i51)